### DISCUSSION TITLE
How to add authentication for accessing the OpenAPI spec file ?

### DISCUSSION THREAD

**Discussion by manbeing** [apiflask/apiflask#274] [Q&A]
I would like to add authentication, eg, basic authentication to access the OpenAPI spec file, eg, localhost:5000/openapi.json
I read the help doc and I was able to add authentication to the endpoints/paths, but no ideas how to add authentication to the spec file above.
`@app.auth_required(auth) `can only be used to protect methods/paths.

Thakns.

**Comment by greyli** [comment]
This is not supported yet. I have no idea how to implement that currently, but you can create a feature request issue [here](https://github.com/apiflask/apiflask/issues/new?assignees=&labels=feature&template=feature-request.md&title=).

**Comment by greyli** [comment]
Oh, I got an idea. You can create a view to serve the spec file by yourself, then add auth support on this view:

```py
@app.route('/my-openapi.json')
def get_spec():
    return app._get_spec()
```

Not tested yet. Feedbacks are welcome!
